src/cplscheme/BaseCouplingScheme.hpp

```diff
@@ -545,6 +476,88 @@ class BaseCouplingScheme : public CouplingScheme {
 
   /// Writes out coupling convergence within all time windows.
   std::shared_ptr<io::TXTTableWriter> _convergenceWriter;
+
+  /// First participant name.
+  std::string _firstParticipant = "unknown";
+
+  /// Second participant name.
+  std::string _secondParticipant = "unknown";
+
+  /// Local participant name.
+  std::string _localParticipant = "unknown";
+
+  /// Smallest number, taking validDigits into account: eps = std::pow(10.0, -1 * validDigits)
+  const double _eps;
+
+  /**
+   * @brief All convergence measures of coupling iterations.
+   *
+   * Before initialization, only dataID and measure variables are filled. Then,
+   * the data is fetched from send and receive data assigned to the cpl scheme.
+   */
+  std::vector<ConvergenceMeasure> _convergenceMeasures;
+
+  /// Sets whether the solver evaluates the fine or the coarse model representation
+  bool _isCoarseModelOptimizationActive = false;
+
+  /// Functions needed for initialize()
+
+  /// implements functionality needed by initialize if __couplingMode == Implicit
+  virtual void initializeImplicit() = 0;
+
+  /// implements functionality for initialize in base class.
+  virtual void initializeImplementation() = 0;
+
+  /// Functions needed for initializeData()
+
+  /// implements functionality for initializeData in base class.
+  virtual void initializeDataImpl() = 0;
```

Comment:
Naming scheme is not consistent. "Impl" vs "Implementation". I guess we could also find better names here in general.

Reply:
The name was only a first shot during refactoring without a lot of thinking about the name.

`InitializeDataImpl()` is called by `initializeData()` as a building block. Therefore, a good name should describe what's happening inside the building block. Since for me this refactoring was mainly a "mechanical" step without looking deeply into `initializeData()` I cannot really provide a good suggestion for a name.

What I understand looking at `initializeData()` and `initializeDataImpl()`:

* `initializeData()` does some checks and sets a flag (`_hasDataBeenExchanged`). So mainly preparation.
* `initializeDataImpl()` does the actual work, which is (from a high-level perspective) very similar for the different `BaseCouplingSchemes`. However, technically the implementations differ. This avoids factoring out further functionality into the base class. **So what does `initializeDataImpl()` do?** Check whether data has to be sent/received, send/receive data and (if necessary) prepare data for extrapolation.

Some name suggestions:

* `exchangeInitialData`
* `doDataInitialization`

Reply:
`exchangeInitialData` is good :+1:

Reply:
Done 4b266ad517b0b4687d5ad2bfef33cf891d4e1dc3.